packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -492,19 +493,22 @@ abstract class ImageStreamCompleter with Diagnosticable {
   /// Adds a listener callback that is called whenever a new concrete [ImageInfo]
   /// object is available or an error is reported. If a concrete image is
   /// already available, or if an error has been already reported, this object
-  /// will notify the listener synchronously.
+  /// will notify the listener synchronously if [synchronousCall] is set to true.
+  ///
+  /// The [synchronousCall] parameter can be used to indicate when this method has
+  /// not been called synchronously. It defaults to true.
   ///
   /// If the [ImageStreamCompleter] completes multiple images over its lifetime,
   /// this listener's [ImageStreamListener.onImage] will fire multiple times.
   ///
   /// {@macro flutter.painting.imageStream.addListener}
-  void addListener(ImageStreamListener listener) {
+  void addListener(ImageStreamListener listener, [bool synchronousCall = true]) {
     _checkDisposed();
     _hadAtLeastOneListener = true;
     _listeners.add(listener);
     if (_currentImage != null) {
       try {
-        listener.onImage(_currentImage!.clone(), true);
+        listener.onImage(_currentImage!.clone(), synchronousCall);
```

Comment:
You can add a private flag in ImageStreamCompleter to indicate whether you are adding an initial listener, and set the synchronousCall based on the flag.

This avoids the breaking change while still make sure the initial listener is added through this method for subclass.